xbmc/dialogs/GUIDialogMediaSource.cpp

```diff
@@ -395,6 +396,12 @@ void CGUIDialogMediaSource::OnOK()
     {
       CGUIWindowVideoBase::OnAssignContent(share.strPath);
     }
+    else if (m_type == "music" &&
```

Comment:
I would re-order this to have a `if` condition that checks for `rss://` and `upnp://` and then inside that block the two different conditions for video and music to save some code duplication.